File: ng2-dart/lib/pipe/comment.pipe.dart
import 'package:angular2/core.dart';

import '../model/activity.dart';

@Pipe(	
	name: 'CommentPipe',
  	pure: false
)
class CommentPipe extends PipeTransform {

  transform(Activity act) {
  	if (act != null) {
  		return act.type == "COMMENT";
	} else {
		return false;
  	}
  }
}

File: ng2-ts/src/app/pipe/comment.pipe.ts
import { Pipe } from '@angular/core';
import { Activity } from '../model/activity';


// Tell Angular2 we're creating a Pipe with TypeScript decorators
@Pipe({
  name: 'CommentPipe',
  pure: false
})
export class CommentPipe {

  transform(value:any) {
  	if (value != null) {
    	return value.filter((activity:Activity) => {
      	return activity.type == 'COMMENT';
    	});
	  } else {
		  return null;
  	}
  }
}